src/components/reviews/Reviews.jsx

```diff
@@ -1,73 +1,111 @@
+      reviewsPerLoad: 2,
+      reviewCount: 0,
+      moreToLoad: true,
+      isLoading: false
     };
-    this.loadReviews = this.loadReviews.bind(this);
+    this.loadMoreReviews = this.loadMoreReviews.bind(this);
     this.setFilter = this.setFilter.bind(this);
   }
 
   componentDidMount() {
-    this.loadReviews(1, 2);
+    this.fetchReviews(1, this.state.reviewsPerLoad);
   }
 
-  loadReviews(page, count) {
-    const intializationConfig = {
-      method: 'get',
-      url: 'https://app-hrsei-api.herokuapp.com/api/fec2/hr-nyc/products',
-      headers: {
-        Authorization: API_KEY,
-      },
-      params: {
-        page,
-        count,
-        'product_id': this.props.productId,
-        'sort': 'newest'
-      }
-    };
+  fetchReviews(page, count, reviewsList = [], callback = null) {
+    const { productId } = this.props;
+    const { isLoading } = this.state;
+    if (productId && !isLoading) {
+      const intializationConfig = {
+        method: 'get',
+        url: 'https://app-hrsei-api.herokuapp.com/api/fec2/hr-nyc/reviews',
+        headers: {
+          Authorization: API_KEY,
+        },
+        params: {
+          page,
+          count,
+          'sort': 'newest',
+          'product_id': this.props.productId
+        }
+      };
+
+      this.setState({ isLoading: true });
+      axios(intializationConfig)
+        .then((response) => {
+          this.setAndFilterReviews(reviewsList, response.data.results, callback);
+        })
+        .catch((error) => {
+          console.log(error);
+        });
+    }
+  }
 
-    axios(intializationConfig)
-      .then((response) => {
-        this.setAndFilterReviews(response.data);
-      })
-      .catch((error) => {
-        console.log(error);
-      });
+  loadMoreReviews(callback = null) {
+    const { reviewCount, reviewsList, reviewsPerLoad } = this.state;
+    const page = (reviewCount + reviewsPerLoad) / reviewsPerLoad;
+    this.fetchReviews(page, reviewsPerLoad, reviewsList, callback);
   }
 
   // setFilter ( function `filter` )
   // sets the review filter to the input function, then refreshes the displayed reviews
-  setFilter(filter) {
+  setFilter(filter, callback = null) {
     this.setState({ filter }, () => {
-      this.setAndFilterReviews(this.state.reviewsList);
+      this.setAndFilterReviews(this.state.reviewsList, [], callback);
     });
   }
 
-  setAndFilterReviews(reviewsList) {
+  setAndFilterReviews(oldReviews, newReviews, callback = null) {
+    const reviewIds = {}; // keep track of IDs already in the list to eliminte duplicates
+    const reviewsList = [];
+    for (let review of oldReviews) {
+      reviewsList.push(review);
+      reviewIds[review.review_id] = true;
+    }
+    for (let review of newReviews) {
+      if (!reviewIds[review.review_id]) {
+        reviewsList.push(review);
+      }
+    }
+    const reviewCount = reviewsList.length;
+    const moreToLoad = newReviews.length === this.state.reviewsPerLoad;
```

Comment:
It seems like the 'moreToLoad' is for the button that user clicks, if there's still unloaded reviews. But how can this be false when the user finish reading every reviews?
The 'reviewsPerLoad' looks like it's 2 and I wasn't sure how the newReviews.length can be changed..

Reply:
reviewsPerLoad is how many new reviews show up every time you click the button. newReviews is the list of reviews that just got sent back from the server. So if the number of reviews you get back is less than you expected, you know that you must be out of reviews, and so there are no more reviews to load.

Reply:
As I was typing this comment I realized there was a bug in this implementation where the button wouldn't disappear immediately if there is an even amount of reviews... but honestly that's not a very serious bug, so I'm gonna leave it for now and make a ticket to figure out a fix tomorrow.